Segment anonymousId backfilling section updates (#392)
rudderlabs/rudderstack-docs: user-guides/migration-guides/rudderstack-migration-guide.mdx

---
# slug: "/docs/user-guides/migration-guides/rudderstack-migration-guide"
title: "Migrating from Segment to RudderStack"
description: Step-by-step guide on migrating from Segment to RudderStack.
---

# Migrating from Segment to RudderStack

> This document explains the step-by-step process of migrating from [Segment](https://segment.com/) to RudderStack. Our primary goal is to lay out the necessary actions for replacing your instrumentation code for generating the events from using the Segment SDK to RudderStack SDK with minimal changes.

Reviewer: ```suggestion
This document explains the step-by-step process of migrating from [Segment](https://segment.com/) to RudderStack. The primary goal is to lay out the necessary actions for replacing your instrumentation code and start generating events from RudderStack SDK with minimal changes.

```

## Migrating the workspace

> Start with creating an account in [RudderStack](https://app.rudderstack.com/signup?type=freetrial). You will then see the following dashboard containing your data plane URL:

Reviewer: ```suggestion
After creating a [RudderStack](https://app.rudderstack.com/signup?type=freetrial) account, you will see the following dashboard containing your data plane URL:
```

---
# slug: "/docs/user-guides/migration-guides/rudderstack-migration-guide"
title: "Migrating from Segment to RudderStack"

> description: Step-by-step guide to migrate from Segment to RudderStack.

Reviewer 2: I don't think "to" is correct here. "Step-by-step guide on migrating from Segment to RudderStack." would be better.

Author: Agree - changed this to your suggestion.

---

# Migrating from Segment to RudderStack

This document explains the step-by-step process of migrating from [Segment](https://segment.com/) to RudderStack. Our primary goal is to lay out the necessary actions for replacing your instrumentation code for generating the events from using the Segment SDK to RudderStack SDK with minimal changes.

## Migrating the workspace

> Start with creating an account in [RudderStack](https://app.rudderstack.com/signup?type=freetrial). You will then see the following dashboard containing your data plane URL required to [set up your data plane](#setting-up-the-rudderstack-data-plane):

Reviewer 2: This is not correct. Data plane URL is not required to set up your data plane. The data plane is already set up.

Author: Makes sense. Removed this section.

<img src="../../assets/user-guides/rudderstack-dashboard.png" alt="RudderStack dashboard" />

Similar to Segment, you will need to set up your source-destination connections in the dashboard and facilitate the event data flow.

<div class="successBlock">

Keep your source write key and the data plane URL handy.
</div>

<div class="infoBlock">

Follow our guide on <a href="https://rudderstack.com/docs/connections/adding-source-and-destination-rudderstack/">Adding a Source and Destination in RudderStack</a> for more information.
</div>

> ## Setting up the RudderStack data plane

Reviewer 2: As I said in my previous comment, the data plane is already set up if they are using a free tier account. Setting up the data plane only makes sense for open source users.

Author: Agreed. Removed it from the doc entirely.

</ul>
<div class="successBlock">

>   You can use the rest of your code as is, as RudderStack SDK API is fully compatible

Reviewer 2: Sounds a little weird saying "as is, as."
Should be "**the** RudderStack SDK API."

Author: You're right - rephrased it accordingly.

			}
		} (method)
	}
	rudderanalytics.load(<WRITE_KEY>, <DATA_PLANE_URL>);
	rudderanalytics.page();
</script>
<script src="https://cdn.rudderlabs.com/v1/rudder-analytics.min.js"></script>
```
</span>
<div class="infoBlock">

For the minified version of the above script, refer to the <a href="https://www.rudderstack.com/docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/quick-start-guide/">RudderStack JavaScript SDK</a> guide.
</div>
</li>
<li>Update the object. RudderStack uses <code class="inline-code">rudderanalytics</code> as the global object library as compared to Segment's <code class="inline-code">analytics</code> object.

<div class="successBlock">

  You can use the rest of your code as is, as the RudderStack SDK API is fully compatible with Segment.
</div>
</li>
</ul>
      </TabPanel>
    </TabPanels>
</Tabs>

## Backfilling Segment anonymous IDs

> When migrating from Segment or a similar analytics tool, you might already have some anonymous traffic that has not yet been identified. When Segment or RudderStack track events for non-identified users, both assign a random UUID as an `anonymousId`. This ID is used to track an unknown user until they are identified. It also allows RudderStack to track user behavior, journeys, and first-touch attribution before and after the users are identified.

Reviewer 2: "When Segment **and** RudderStack track events" might be better.
